client/mm/libxc/bitget.go

```
@@ -0,0 +1,3133 @@
+	} else if status.Complete {
+		return nil // Order is already complete, no need to cancel
+	}
+
+	symbol := bitgetMktID(baseCfg, quoteCfg)
+
+	cancelReq := &bgtypes.CancelOrderRequest{
+		Symbol:    symbol,
+		ClientOid: tradeID,
+	}
+
+	var resp bgtypes.APIResponse
+	err = bg.postAPI(ctx, "/api/v2/spot/trade/cancel-order", cancelReq, true, &resp)
+	if err != nil {
+		return err
+	}
+
+	if resp.Code != "00000" {
+		return fmt.Errorf("bitget API error: %s - %s", resp.Code, resp.Msg)
+	}
+
+	return nil
+}
+
+// TradeStatus returns the status of a trade on Bitget
+func (bg *bitget) TradeStatus(ctx context.Context, tradeID string, baseID, quoteID uint32) (*Trade, error) {
+	baseCfg, err := bitgetAssetCfg(baseID)
+	if err != nil {
+		return nil, err
+	}
+
+	quoteCfg, err := bitgetAssetCfg(quoteID)
+	if err != nil {
+		return nil, err
+	}
+
+	symbol := bitgetMktID(baseCfg, quoteCfg)
+
+	params := map[string]string{
+		"symbol":    symbol,
+		"clientOid": tradeID,
+	}
+
+	var resp bgtypes.APIResponse
+	err = bg.getAPI(ctx, "/api/v2/spot/trade/orderInfo", params, true, &resp)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.Code != "00000" {
+		return nil, fmt.Errorf("bitget API error: %s - %s", resp.Code, resp.Msg)
+	}
+
+	// Bitget API returns an array of orders, even when querying by clientOid
+	var orderDetails []bgtypes.OrderDetail
+	if err := json.Unmarshal(resp.Data, &orderDetails); err != nil {
+		return nil, fmt.Errorf("error unmarshaling order detail: %w", err)
+	}
+
+	if len(orderDetails) == 0 {
+		return nil, fmt.Errorf("order not found: %s", tradeID)
+	}
+
+	orderDetail := orderDetails[0]
+
+	sell := orderDetail.Side == "sell"
+	market := orderDetail.OrderType == "market"
+
+	// Parse quantities
+	size, _ := strconv.ParseFloat(orderDetail.Size, 64)
+	baseFilled, _ := strconv.ParseFloat(orderDetail.BaseVolume, 64)
+	quoteFilled, _ := strconv.ParseFloat(orderDetail.QuoteVolume, 64)
+	priceAvg, _ := strconv.ParseFloat(orderDetail.PriceAvg, 64)
```

Rather not ignore errors.